### ISSUE TITLE
 Will download and extract kubernetes-test tarball(s) -  curl: (22) The requested URL returned error: 403 

### ISSUE THREAD

**Issue by droslean** [kubernetes/kubernetes#88367]
Many jobs are failing because of 
```console
 I0220 10:10:17.124] Will download and extract kubernetes-test tarball(s) from https://storage.googleapis.com/kubernetes-release-dev/ci/v1.19.0-alpha.0.64+937008e3aca75c
I0220 10:10:17.124] https://www.googleapis.com/auth/cloud-platform
W0220 10:10:17.225]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
W0220 10:10:17.225]                                  Dload  Upload   Total   Spent    Left  Speed
W0220 10:10:17.328] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
W0220 10:10:17.328] curl: (22) The requested URL returned error: 403 
```

Some of the job logs:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-ingress/1230434368032346114
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-ingress/1230434368032346114
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/1230440661795213312#1:build-log.txt%3A205
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-gce-conformance-latest/1230420526376161281#1:build-log.txt%3A161

/priority critical-urgent
/cc @kubernetes/ci-signal 
/cc @justaugustus @alejandrox1 @kubernetes/release-managers 
/milestone v1.18

/sig release

**Comment by droslean** [comment]
Also, all the jobs report in their log the following:
```
bootstrap.py is deprecated!
test-infra oncall does not support any job still using bootstrap.py.
Please migrate your job to podutils!
https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md
```

@justaugustus  Do we have to take any action to change that?

**Comment by droslean** [comment]
After some investigation, it seems that file `gs://kubernetes-release-dev/ci/latest-green.txt` contains `v1.19.0-alpha.0.47+da9f47eadfdf85` instead of `v1.19.0-alpha.0.64+937008e3aca75c`

**Comment by alejandrox1** [comment]
> Also, all the jobs report in their log the following:
> 
> ```
> bootstrap.py is deprecated!
> test-infra oncall does not support any job still using bootstrap.py.
> Please migrate your job to podutils!
> https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md
> ```
> 
> @justaugustus Do we have to take any action to change that?

Switching from bootstrap.py to podutils should be done by the job owners - this is out of scope for releas team (strictly speaking).
And the message is just a warning, should not affect the job itself.

**Comment by liggitt** [comment]
@BenTheElder @fejta is this related to the workload identity switch yesterday?

**Comment by BenTheElder** [comment]
Maybe? get-kube.sh does seem to pass auth headers, possibly this part isn't working correctly?
These jobs explicitly mount a service-account.json though so I wouldn't expect workload identity to change things but I could be wrong...

https://github.com/kubernetes/kubernetes/blob/afa72e93cd55ec4f873b8d218054fc8b5f060e02/cluster/get-kube.sh#L246

**Comment by fejta** [comment]
Yes, this will not work until we attach a kubernetes service account to the job that uses workload identity:

https://github.com/kubernetes/kubernetes/blob/afa72e93cd55ec4f873b8d218054fc8b5f060e02/cluster/get-kube.sh#L125-L130

**Comment by fejta** [comment]
Ideally that command should become `gcloud auth print-access-token` (and let it decide how to best authenticate) rather than assuming the script runs on a VM

**Comment by BenTheElder** [comment]
sent https://github.com/kubernetes/kubernetes/pull/88383

**Comment by droslean** [comment]
/reopen

#88383 doesn't seem to solve the issue. The jobs are still failing because of this reason. 

@BenTheElder Do we need any more actions after #88383 ????

**Comment by mm4tt** [comment]
> /reopen
> 
> #88383 doesn't seem to solve the issue. The jobs are still failing because of this reason.
> 
> Benjamin Elder Do we need any more actions after #88383 ????

I don't agree, the https://github.com/kubernetes/kubernetes/pull/88383 definitely helped in case of e2e scalability jobs, our presubmits and continuous jobs started passing again, e.g. 
https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-cos-master-scalability-100

**Comment by BenTheElder** [comment]
https://prow.k8s.io/job-history/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-ingress

Looks green?

**Comment by droslean** [comment]
@mm4tt  @BenTheElder Only the 1.18 jobs are green. We need to cherry-pick the fix to the other branches as well. I will keep this issue open.

https://testgrid.k8s.io/sig-release-1.17-blocking#gce-cos-k8sstable1-ingress
https://testgrid.k8s.io/sig-release-1.17-blocking#gce-cos-k8sstable1-reboot
https://testgrid.k8s.io/sig-release-1.17-blocking#gce-cos-k8sstable1-alphafeatures
https://testgrid.k8s.io/sig-release-1.17-blocking#Conformance%20-%20GCE%20-%201.17

**Comment by BenTheElder** [comment]
For some reason I thought kubetest always used the copy from HEAD.

**Comment by BenTheElder** [comment]
in the case that it didn't we'll need this cherry picked to all active branches :|

**Comment by mm4tt** [comment]
I thought the same, but indeed the https://github.com/kubernetes/kubernetes/pull/88383 fixed only 1.18 and master jobs. Older release scalability jobs are still broken
![BJtzLzez5Qb](https://user-images.githubusercontent.com/2604887/75092706-c2281800-557a-11ea-9a00-28b7c2bb9d8c.png)

It's a bit confusing TBH. I thought that get-kube.sh is always downloaded by kubetest from https://get.k8s.io/  which IIUC points to the master branch.
Alternatively, get-kube.sh could've been embedded in the kubekins image, but if that was the case then https://github.com/kubernetes/kubernetes/pull/88383 couldn't have fixed our 1.18/master tests without kubekins version update (which didn't happen).
Another options is that kubetest downloads get-kube.sh from proper release branches, but I couldn't find such logic in the kubetest code.

@krzyzacy, @fejta would you mind explaining how it works?

**Comment by droslean** [comment]
There are a lot of jobs that are failing because of this.

[skew-cluster-latest-kubectl-stable1-gce](https://testgrid.k8s.io/sig-release-master-blocking#skew-cluster-latest-kubectl-stable1-gce) - master
[skew-cluster-latest-kubectl-beta-gce](https://testgrid.k8s.io/sig-release-1.18-blocking#skew-cluster-latest-kubectl-beta-gce) - 1.18

7 jobs in 1.17 branch https://testgrid.k8s.io/sig-release-1.17-blocking
5 jobs in 1.16 branch https://testgrid.k8s.io/sig-release-1.16-blocking
5 jobs in 1.15 branch https://testgrid.k8s.io/sig-release-1.15-blocking

@krzyzacy @fejta @BenTheElder Can you please take a look?

**Comment by mm4tt** [comment]
Ping, any updates?

**Comment by BenTheElder** [comment]
@spiffxp filed a number of cherry picks for the script

**Comment by droslean** [comment]
Closing... The issue has been resolved. Thanks, @BenTheElder @spiffxp @mm4tt 
/close
